core/state_processor.go

```diff
@@ -73,6 +74,8 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 			return nil, nil, 0, err
 		}
 	}
+	blockchain_parameters.CheckMinimumVersion(header, statedb)
```

Comment:
Is this the right place to check this? I think we process all proposed blocks before they're finalized.

Since it's not critical that we shut down out-of-date node immediately, consider instead a separate go routine that just checks the version against the latest finalized block every few minutes.